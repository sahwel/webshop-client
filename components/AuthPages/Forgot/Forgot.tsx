import React from 'react';
import AuthPagesContainer from '../AuthPagesContainer';
import ForgotFooter from './Components/ForgotFooter';
import ForgotHeader from './Components/ForgotHeader';
import ForgotInputs from './Components/ForgotInputs';

const Forgot = () => {
  return (
    <AuthPagesContainer>
      <form className="pt-20 grid gap-16 ">
        <ForgotHeader />
        <ForgotInputs />
        <ForgotFooter />
      </form>
    </AuthPagesContainer>
  );
};

export default Forgot;
