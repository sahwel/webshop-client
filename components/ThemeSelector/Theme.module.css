
@tailwind components;

@layer components {
    .theme {
       @apply rounded-full p-2  flex justify-center items-center
    }
}