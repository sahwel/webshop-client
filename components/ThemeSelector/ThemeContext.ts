/* eslint-disable no-undef */
import { createContext } from 'react';
import { Themes } from './definitions';

export interface ThemeContextModel {
  setTheme(): void;
  getTheme(): Themes;
}

export const ThemeContext = createContext<ThemeContextModel>(null as any);
