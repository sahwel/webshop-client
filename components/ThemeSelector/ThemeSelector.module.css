
@tailwind components;

@layer components {
    .theme_sececltor {
        @apply grid grid-cols-2 z-50 cursor-pointer bg-white absolute right-5  rounded-full border-2 border-theme-border-light
    }  
    
    .theme_sun{
        @apply text-orange-600 
    }

    .theme_moon{
        @apply text-gray-400 rotate-[255deg]
    }

    .theme_moon_highlited{
        @apply bg-[#303841]
    }

    .theme_sun_highlited{
        @apply bg-orange-200
    }
}