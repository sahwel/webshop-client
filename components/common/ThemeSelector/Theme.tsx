import { FunctionComponent, ReactNode } from 'react';
import cl from 'classnames';
import styles from './theme.module.css';

interface ThemeProps {
  children?: ReactNode;
  isHighlighted: boolean;
  highlightedClass: string;
}

const Theme: FunctionComponent<ThemeProps> = ({
  children,
  isHighlighted,
  highlightedClass,
}) => {
  return (
    <div className={cl({ [highlightedClass]: isHighlighted }, styles.theme)}>
      {children}
    </div>
  );
};

export default Theme;
