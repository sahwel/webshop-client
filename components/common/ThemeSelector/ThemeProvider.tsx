import {
  FunctionComponent,
  ReactNode,
  useContext,
  useEffect,
  useLayoutEffect,
  useMemo,
  useState,
} from 'react';
import { ThemeContext, ThemeContextModel } from './ThemeContext';
import { Themes } from './definitions';

interface ThemeProviderProps {
  children?: ReactNode;
}

const ThemeProvider: FunctionComponent<ThemeProviderProps> = ({ children }) => {
  const [currentTheme, setCurrentTheme] = useState<Themes>('light');

  useLayoutEffect(() => {
    // todo: get alternative solution
    const theme = localStorage.theme ? localStorage.theme : 'light';

    if (theme === 'dark') {
      document.documentElement.classList.add('dark');
    } else {
      document.documentElement.classList.remove('dark');
    }
    setCurrentTheme(theme);

    return () => {};
  }, []);
  const contextValue: ThemeContextModel = useMemo(
    () => ({
      setTheme() {
        setCurrentTheme((oldTheme) => {
          const newTheme = oldTheme === 'light' ? 'dark' : 'light';

          if (newTheme === 'dark') {
            document.documentElement.classList.add('dark');
          } else {
            document.documentElement.classList.remove('dark');
          }
          localStorage.theme = newTheme;
          return newTheme;
        });
      },
      getTheme() {
        return currentTheme;
      },
    }),
    [currentTheme]
  );

  return (
    <ThemeContext.Provider value={contextValue}>
      {children}
    </ThemeContext.Provider>
  );
};

export default ThemeProvider;

export function useTheme() {
  const value = useContext(ThemeContext);
  return value;
}
