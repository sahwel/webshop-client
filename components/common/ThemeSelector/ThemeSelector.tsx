import { SunIcon } from '@heroicons/react/outline';
import { MoonIcon } from '@heroicons/react/solid';
import { useCallback, useLayoutEffect, useState } from 'react';
import cl from 'classnames';
import Theme from './Theme';
import styles from './ThemeSelector.module.css';
import { useTheme } from './ThemeProvider';

const ThemeSelector = () => {
  const siteTheme = useTheme();
  const handleChange = useCallback(() => {
    siteTheme.setTheme();
  }, []);

  return (
    <div className={styles.theme_sececltor} onClick={handleChange}>
      <Theme
        highlightedClass={styles.theme_sun_highlited}
        isHighlighted={siteTheme.getTheme() === 'light'}
      >
        <SunIcon className={cl('icon-6', styles.theme_sun)} />
      </Theme>

      <Theme
        highlightedClass={styles.theme_moon_highlited}
        isHighlighted={siteTheme.getTheme() === 'dark'}
      >
        <MoonIcon
          className={cl('icon-6', styles.theme_moon, {
            ['!text-white ']: siteTheme.getTheme() === 'dark',
          })}
        />
      </Theme>
    </div>
  );
};

export default ThemeSelector;
