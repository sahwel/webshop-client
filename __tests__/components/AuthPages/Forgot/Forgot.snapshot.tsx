import { render, renderHook, RenderHookResult } from '@testing-library/react';
import { useTheme } from '@/components/common/ThemeSelector/ThemeProvider';
import {
  mockUseThemeSelector,
  themeSelectorWrapper,
} from '../../../../testUtils/MockThemeSelector';
import React from 'react';
import Forgot from '@/pages/forgot';
React.useContext = mockUseThemeSelector;

describe('Forgot Snapshot', () => {
  let result: RenderHookResult<any, any>;
  beforeEach(() => {
    result = renderHook(() => useTheme(), {
      wrapper: themeSelectorWrapper,
    });
  });

  it('renders Forgot unchanged', () => {
    const { container } = render(<Forgot />);
    expect(container).toMatchSnapshot();
  });
});
