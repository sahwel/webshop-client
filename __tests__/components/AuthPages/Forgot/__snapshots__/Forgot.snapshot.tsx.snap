// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forgot Snapshot renders Forgot unchanged 1`] = `
<div>
  <div
    class="auth_pages_container"
  >
    <div
      class="blob_container"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="blob"
          class="blob"
          data-nimg="fill"
          decoding="async"
          sizes="100vw"
          src="/blobs/light/login-register/blob.svg"
          srcset="/blobs/light/login-register/blob.svg 640w, /blobs/light/login-register/blob.svg 750w, /blobs/light/login-register/blob.svg 828w, /blobs/light/login-register/blob.svg 1080w, /blobs/light/login-register/blob.svg 1200w, /blobs/light/login-register/blob.svg 1920w, /blobs/light/login-register/blob.svg 2048w, /blobs/light/login-register/blob.svg 3840w"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </span>
    </div>
    <form
      class="pt-20 grid gap-16 "
    >
      <h1
        class="text-[300%]"
      >
        Create your new password
      </h1>
      <div
        class="input_container"
      >
        <p>
          Enter the email associated with your account and weâ€™ll send you an email with instructions to reset your password
        </p>
        <div
          class="input_container"
        >
          <input
            class="input"
            placeholder="Email"
            type="text"
          />
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-envelope input_icon icon-7"
            data-icon="envelope"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 128C0 92.65 28.65 64 64 64H448C483.3 64 512 92.65 512 128V384C512 419.3 483.3 448 448 448H64C28.65 448 0 419.3 0 384V128zM48 128V150.1L220.5 291.7C241.1 308.7 270.9 308.7 291.5 291.7L464 150.1V127.1C464 119.2 456.8 111.1 448 111.1H64C55.16 111.1 48 119.2 48 127.1L48 128zM48 212.2V384C48 392.8 55.16 400 64 400H448C456.8 400 464 392.8 464 384V212.2L322 328.8C283.6 360.3 228.4 360.3 189.1 328.8L48 212.2z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
      <div>
        <button
          class="button"
          type="submit"
        >
          Send email
        </button>
        <a
          class="text-center block text-sm mt-2 cursor-pointer"
          href="/login"
        >
          Back
        </a>
      </div>
    </form>
  </div>
</div>
`;
